include/nbl/video/SPhysicalDeviceFeatures.h

```diff
@@ -0,0 +1,775 @@
+
+    /* YcbcrImageArraysFeaturesEXT *//* VK_EXT_ycbcr_image_arrays */
+    //VkBool32           ycbcrImageArrays;
+
+    /* ShaderIntegerFunctions2FeaturesINTEL *//* VK_INTEL_shader_integer_functions2 */
+    //VkBool32           shaderIntegerFunctions2;
+
+    /* 16BitStorageFeaturesKHR *//* VK_KHR_16bit_storage *//* MOVED TO Vulkan 1.1 Core */
+    /* 8BitStorageFeaturesKHR *//* VK_KHR_8bit_storage *//* MOVED TO Vulkan 1.2 Core */
+
+    /* AccelerationStructureFeaturesKHR *//* VK_KHR_acceleration_structure */
+    bool accelerationStructure = false;
+    bool accelerationStructureCaptureReplay = false;
+    bool accelerationStructureIndirectBuild = false;
+    bool accelerationStructureHostCommands = false;
+    bool descriptorBindingAccelerationStructureUpdateAfterBind = false;
+            
+    /* BufferDeviceAddressFeaturesKHR *//* VK_KHR_buffer_device_address *//* MOVED TO Vulkan 1.2 Core */
+
+    /* DynamicRenderingFeaturesKHR *//* VK_KHR_dynamic_rendering *//* MOVED TO Vulkan 1.3 Core */
+
+    /* [!!NV Version below, struct doesn't exist in vk headers] VK_KHR_fragment_shader_barycentric */
+    
+    // [DO NOT EXPOSE] not implementing or exposing VRS in near or far future
+    /* FragmentShadingRateFeaturesKHR *//* VK_KHR_fragment_shading_rate */
+    //VkBool32           pipelineFragmentShadingRate;
+    //VkBool32           primitiveFragmentShadingRate;
+    //VkBool32           attachmentFragmentShadingRate;
+
+    /* ImagelessFramebufferFeaturesKHR *//* VK_KHR_imageless_framebuffer *//* MOVED TO Vulkan 1.2 Core */
+    /* Maintenance4FeaturesKHR *//* VK_KHR_maintenance4 *//* MOVED TO Vulkan 1.3 Core */
+    /* MultiviewFeaturesKHR *//* VK_KHR_multiview *//* MOVED TO Vulkan 1.1 Core */
+    
+    // [TODO]
+    /* PerformanceQueryFeaturesKHR *//* VK_KHR_performance_query */
+    //VkBool32           performanceCounterQueryPools;
+    //VkBool32           performanceCounterMultipleQueryPools;
+
+    // [TODO]
+    /* PipelineExecutablePropertiesFeaturesKHR *//* VK_KHR_pipeline_executable_properties */
+    //VkBool32           pipelineExecutableInfo;
+    
+    /* VK_KHR_portability_subset - PROVISINAL/NOTAVAILABLEANYMORE */
+
+    // [DO NOT EXPOSE] no point exposing until an extension more useful than VK_KHR_present_wait arrives
+    /* PresentIdFeaturesKHR *//* VK_KHR_present_id */
+    //VkBool32           presentId;
+    
+    // [DO NOT EXPOSE] won't expose, this extension is poop, I should have a Fence-andQuery-like object to query the presentation timestamp, not a blocking call that may unblock after an arbitrary delay from the present
+    /* PresentWaitFeaturesKHR *//* VK_KHR_present_wait */
+    //VkBool32           presentWait;
+    
+    /* RayQueryFeaturesKHR *//* VK_KHR_ray_query */
+    bool rayQuery = false;
+
+    /* VK_KHR_ray_tracing !! Replaced/Removed */
+    /* VK_KHR_ray_tracing_maintenance1 *//* added in vk 1.3.213, the SDK isn't released yet at this moment :D */
```

Comment:
https://renderdoc.org/vkspec_chunked/chap50.html#_description_26

Reply:
Baiscally:
- Support for `GLSL_EXT_ray_cull_mask`, lets call it `rayCullMask`
- new pipeline stage and access masks but only in `KHR_synchronization2` which we don't use
- two new acceleration structure query parameters
- `rayTracingPipelineTraceRaysIndirect2` feature, same as `rayTracingPipelineTraceRaysIndirect` but with indirect SBT and dispatch dimensions

Lets have
```cpp
bool accelerationStructureSizeAndBLASPointersQuery = false;

// Do not expose, we don't use KHR_synchronization2 yet
//bool accelerationStructureCopyStageAndSBTAccessType;

bool rayCullMask = false;

bool rayTracingPipelineTraceRaysIndirectDimensionsAndSBT = false;
```

Lets enable `rayTracingMaintenance1` and `rayTracingPipelineTraceRaysIndirect2` whenever required by the above.